experiments/test_helpers.rb

```diff
@@ -116,25 +123,33 @@ def start_threads
         @state_transitions = []
         @state_transitions_mutex = Mutex.new
 
-        start_request_threads
+        start_request_arrival
+        start_processing_threads
         subscribe_to_metrics if @is_adaptive
         subscribe_to_state_changes unless @is_adaptive
       end
 
-      def start_request_threads
-        total_rps = @num_threads * @requests_per_second_per_thread
-        puts "Starting #{@num_threads} concurrent request threads (#{@requests_per_second_per_thread} requests/second each = #{total_rps} rps total)..."
-        puts "Each thread will have its own adapter instance connected to the shared service...\n"
+      def start_request_arrival
```

Comment:
In a real service, requests are received externally, and processed by the workers. This implementation is much closer to this reality, and it allows us better to control how many requests we are seeing